halo2/src/halo2.rs

```diff
@@ -5364,7 +5367,7 @@ pub fn check_not_zero_polyx(p: Polyx) -> bool {
 ///
 /// * `p` - the polynomial
 pub fn trim_polyx(p: Polyx) -> Polyx {
-    let mut last_val_idx = 0;
+    let mut last_val_idx = 1;
```

Comment:
I don't think this change is correct ...

Reply:
While it is correct that this change caused the error, this is a question of how we want to define the 0-polynomial. Changing this to 1, defines the 0-polynomial as `Seq { b: [0] }` instead of `Seq { b: [] }`. I think this is a better design. The error was also caused by me forgetting to change an arbitrary implementation for polynomials, which allowed the creation of 0-polynomials of the form `Seq { b: [] }`. This has been fixed now, and the tests that failed in the last CI.

Reply:
Good point. Thanks for the fix!